content/sensu-core/2.0/reference/assets.md

```diff
@@ -1,92 +1,101 @@
-asset, we'll need a name, a URL to the asset location, and a SHA-512 checksum.
-{{< highlight shell >}}
-sensuctl asset create readme \
-  -u http://example.com/README.md \
-  --sha512 "$(sha512sum README.md | cut -f1 -d ' ')"
-OK
-{{< /highlight >}}
-
-### Getting information about an asset
-
-Getting information about an asset is easy - you just need the name.
-
-{{< highlight shell >}}
-sensuctl asset info readme
-=== readme
-Name:             readme
-Organization:     default
-URL:              http://example.com/README.md
-SHA-512 Checksum: 84998adb7e1ced02b092347eb050018c3abe8c0055797ac149a1c3cf9efe6cfc7e9666ffba748729ed0273265aeacfe61b295e1ebfd739e33a124883f7703f0b
-Filters:          
-Metadata:         
-{{< /highlight >}}
-
-### Listing all available assets
-
-{{< highlight shell >}}
-sensuctl asset list
-   Name                 URL                Hash    
- ───────── ───────────────────────────── ───────── 
-  foo/bar   //example.com/.../            asldkjf  
-  foobar    //example.com/.../            0ea6f27  
-  readme    //example.com/.../README.md   84998ad  
-{{< /highlight >}}
-
-### Updating an existing asset
-
-{{< highlight shell >}}
-sensuctl asset update readme
-{{< /highlight >}}
-
-This will result in an interactive prompt, which will ask you which of the
-asset's field that you wish to change.
-
-## Using assets in checks
-
-Assets can be associated with checks when they are created, or afterwards.
-
-### Adding an asset to a check on creation
-
-{{< highlight shell >}}
-sensuctl check create foobar \
-  --command /bin/false \
-  --subscriptions foobar \
-  --interval 10 \
-  --runtime-assets readme,foobar
-{{< /highlight >}}
-
-### Adding an asset to an existing check's dependencies
-
-{{< highlight shell >}}
-sensuctl check set-runtime-assets check1 readme,foobar
-{{< /highlight >}}
-
-This command will set a check's assets to `readme` and `foobar`.
+## How do assets work?
+At runtime, the agent sequentially fetches assets and stores them in its local
+cache. Asset dependencies are then injected into the `PATH` so they are
+available when the command is executed. Subsequent check executions look for the
+asset in local cache and ensure the contents match the checksum. If the
+requested asset is not in the local cache, it is downloaded from the asset url.
+The Sensu backend does not currently provide any storage for assets; they are
+expected to be retrieved over HTTP or HTTPS.
+
+The agent expects that an asset is a `TAR` archive that may optionally be GZip'd.
+Any scripts or executables should be within a `bin/` folder within in the archive.
+
+The following are injected into the execution context:
+
+- `{PATH_TO_ASSET}/bin` is injected into the `PATH` environment variable.
+- `{PATH_TO_ASSET}/lib` is injected into the `LD_LIBRARY_PATH` environment variable.
+- `{PATH_TO_ASSET}/include` is injected into the `CPATH` environment variable.
+
+## Assets specification
+
+### Attributes
+
+name         | 
```

Comment:
I've used the following format, inherited from 1.x on other reference docs: https://docs.sensu.io/sensu-core/2.0/reference/checks/#check-naming

But I think I like your approach better. Could you just add the link to the online regex validator? And we'll switch the rest of the docs to this format!